crowbar_framework/app/controllers/barclamp_controller.rb

```diff
@@ -271,103 +252,279 @@ def get_proposals_from_barclamps(barclamps)
+    code, message = @service_object.proposal_delete(
+      params[:id]
+    )
+
+    raise Crowbar::Error::NotFound if code == 404
+    respond_to do |format|
+      case code
+      when 200
+        format.json do
+          head :ok
+        end
+      else
+        format.json do
+          render json: { error: message }, status: code
+        end
       end
-    rescue StandardError => e
-      flash_and_log_exception(e)
     end
+  end
+
+  #
+  # Commit proposal
+  # Commit a specific proposal to apply it
+  # POST /crowbar/<barclamp-name>/<version>/proposals/commit/<barclamp-instance-name>
+  #
+  add_help(:proposal_commit, [:id], [:post])
+  def proposal_commit
+    code, message = @service_object.proposal_commit(
+      params[:id]
+    )
+
+    raise Crowbar::Error::NotFound if code == 404
     respond_to do |format|
-      format.html {
-        return redirect_to barclamp_modules_path id: params[:controller] if answer[0] != 200
-        redirect_to proposal_barclamp_path controller: controller, id: orig_id
-      }
-      format.xml  {
-        return render text: flash[:alert], status: answer[0] if answer[0] != 200
-        render xml: answer[1]
-      }
-      format.json {
-        return render text: flash[:alert], status: answer[0] if answer[0] != 200
-        render json: answer[1]
-      }
+      case code
+      when 200
+        format.html do
+          redirect_to(
+            proposal_barclamp_path(
+              controller: params[:id],
+              id: params[:id]
+            )
+          )
+        end
+        format.json do
+          head :ok
+        end
+      when 202
+        format.html do
+          flash[:warning] = message
+
+          redirect_to(
+            proposal_barclamp_path(
+              controller: params[:id],
+              id: params[:id]
+            )
+          )
+        end
+        format.json do
+          head :ok
+        end
+      else
+        format.html do
+          flash[:alert] = message
+
+          redirect_to(
+            proposal_barclamp_path(
+              controller: params[:id],
+              id: params[:id]
+            )
+          )
+        end
+        format.json do
+          render json: { error: message }, status: code
+        end
+      end
     end
   end
 
   #
-  # Provides the restful api call for
-  # Edit Proposal Instance 	/crowbar/<barclamp-name>/<version>/propsosals/<barclamp-instance-name> 	POST 	Posting a json document will replace the current proposal
+  # Dequeue proposal
+  # Remove a specific proposal from the queue
+  # DELETE /crowbar/<barclamp-name>/<version>/proposals/dequeue/<barclamp-instance-name>
+  #
+  add_help(:proposal_dequeue, [:id], [:delete])
+  def proposal_dequeue
+    code, message = @service_object.dequeue_proposal(
+      params[:id]
+    )
+
+    raise Crowbar::Error::NotFound if code == 404
+    respond_to do |format|
+      case code
+      when 200
+        format.json do
+          head :ok
+        end
```

Comment:
No html?

Also, in the past, we used to have a flash message ot say it got dequeued.